Problem with Newley released types

- Versions: node@v22.1.0, darwin@23.2.0
- `nodemon -v`: 3.1.1
- Operating system/terminal environment (powershell, gitshell, etc):  MacOs, terminal
- Command you ran: Via api in typescript


### Expected behaviour
When importing nodemon there is a default export for functionality defined in the types.

### Actual behaviour
With the addition of https://github.com/remy/nodemon/pull/2204 there where types added to the project, but no default export types for functionality -> that leads to a not usable nodemon in typescript

### Steps to reproduce
import nodemon from 'nodemon';

->  nodemon is not a module

---

Sure here are the  errors that show up after upgrade: 

Let me know if you want a link to the code.

```bash 
@fiction/core/plugin-env/restart.ts:19:13 - error TS2503: Cannot find namespace 'nodemon'.

19     config: nodemon.Settings
               ~~~~~~~

@fiction/core/plugin-env/restart.ts:65:8 - error TS2339: Property 'on' does not exist on type '(settings: NodemonSettings) => Nodemon'.

65       .on('log', () => {})
          ~~
```

<img width="696" alt="Screenshot 2024-05-29 at 1 37 13 PM" src="https://github.com/remy/nodemon/assets/308134/2ef43065-629c-4ccc-9b71-5640811d4c07">

--- a/index.d.ts
+++ b/index.d.ts
@@ -118,9 +118,9 @@ export interface NodemonConfig {
 }
 
 export interface NodemonSettings extends NodemonConfig {
-  script: string; 
+  script: string;
   ext?: string; // "js,mjs" etc (should really support an array of strings, but I don't think it does right now)
-  events?: { [key: string]: string }; 
+  events?: { [key: string]: string };
   env?: { [key: string]: string };
   exec?: string; // node, python, etc
   execArgs?: string[]; // args passed to node, etc,
@@ -136,4 +136,6 @@ export interface WatchOptions {
   interval: number;
 }
 
-export default function nodemon(settings: NodemonSettings): Nodemon;
+const nodemon: Nodemon = (settings: NodemonSettings): Nodemon => {};
+
+export default nodemon;

